Bugsnag/BugsnagSystemState.m

```diff
@@ -189,27 +190,27 @@ - (instancetype)initWithConfiguration:(BugsnagConfiguration *)config {
             [weakSelf setValue:@NO forAppKey:SYSTEMSTATE_APP_IS_IN_FOREGROUND];
         }];
 #else
-        [center addObserverForName:UIApplicationWillTerminateNotification object:nil queue:nil usingBlock:^(NSNotification * _Nonnull note) {
+        [center addObserverForName:@"UIApplicationWillTerminateNotification" object:nil queue:nil usingBlock:^(NSNotification * _Nonnull note) {
```

Comment:
This feels dangerous, because any typo won't be detected. Could we instead define our own constants in one place, and then in the tests use a macro to check against the UIKit ones? e.g.

```c
#define ASSERT_NOTIFICATION_NAME(name) XCTAssertEqualObjects(BSGUI ## name, UI ## name)
ASSERT_NOTIFICATION_NAME(ApplicationDidBecomeActiveNotification);
ASSERT_NOTIFICATION_NAME(ApplicationDidEnterBackgroundNotification);
```